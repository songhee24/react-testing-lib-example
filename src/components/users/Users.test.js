import { fireEvent, render, screen } from "@testing-library/react";
import axios from "axios";
import { renderWithRouter } from "../../tests/helpers/renderWithRouter";
jest.mock("axios");
describe("USERS TEST", () => {
  let response;

  beforeEach(() => {
    response = {
      data: [
        {
          id: 1,
          name: "Leanne Graham",
        },
        {
          id: 2,
          name: "Ervin Howell",
        },
        {
          id: 3,
          name: "Clementine Bauch",
        },
      ],
    };
  });

  test("renders users test", async () => {
    axios.get.mockReturnValue(response);
    render(renderWithRouter(null, "/users"));
    const users = await screen.findAllByTestId("user-item");
    expect(users.length).toBe(3);
    expect(axios.get).toBeCalledTimes(1);
    screen.debug();
    expect(users).toMatchSnapshot();
  });

  test("redirect to details page test", async () => {
    axios.get.mockReturnValue(response);
    render(renderWithRouter(null, "/users"));
    const users = await screen.findAllByTestId("user-item");
    expect(users.length).toBe(3);
    fireEvent.click(users[0]);
    expect(screen.getByTestId("user-page")).toBeInTheDocument();
  });
});
