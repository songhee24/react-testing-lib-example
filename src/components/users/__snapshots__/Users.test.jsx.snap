// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USERS TEST renders users 1`] = `
Array [
  <div
    data-testid="user-item"
  >
    Leanne Graham
  </div>,
  <div
    data-testid="user-item"
  >
    Ervin Howell
  </div>,
  <div
    data-testid="user-item"
  >
    Clementine Bauch
  </div>,
]
`;

exports[`USERS TEST renders users test 1`] = `
Array [
  <a
    data-testid="user-item"
    href="/users/1"
  >
    Leanne Graham
  </a>,
  <a
    data-testid="user-item"
    href="/users/2"
  >
    Ervin Howell
  </a>,
  <a
    data-testid="user-item"
    href="/users/3"
  >
    Clementine Bauch
  </a>,
]
`;
